src/components/EmailInput.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { TextField } from "@mui/material";
+import { FieldErrors, UseFormRegister } from "react-hook-form";
+
+import { LogInInputs } from "../types/LoginInput";
+
+interface EmailInputProps {
+  register: UseFormRegister<LogInInputs>;
+  errors: FieldErrors<LogInInputs>;
+}
+
+const EmailInput: React.FC<EmailInputProps> = ({ register, errors }) => {
+  return (
+    <TextField
+      label="Email address*"
+      type="email"
+      variant="outlined"
+      {...register("email", {
+        required: true,
+        pattern: {
+          value: /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/,
+          message: "Please enter a valid email address.",
+        },
```

Comment:
Especially with more complex forms you'd very soon want to have a central schema defined on the form itself (e.g. with the resolver property). For this simple form it makes sense though :+1:

Reply:
Great point, makes sense. Would you use a library like Zod for the validation? I came across it in an article, but haven't used it before.